pvlib/iotools/solrad.py

```diff
@@ -91,19 +100,28 @@ def read_solrad(filename):
         widths = WIDTHS
         dtypes = DTYPES
 
+    meta = {}
+
+    if str(filename).startswith('ftp') or str(filename).startswith('http'):
+        response = requests.get(filename)
+        file_buffer = io.StringIO(response.content.decode())
+    else:
+        file_buffer = open(str(filename), 'r')
```

Comment:
I have a mild allergic reaction to seeing `= open(...)` in code, since it means that the file handle will get left open if an unhandled exception occurs before the `.close()` is called.  It's possible to refactor this code to use `with open(...)` instead, but I think it would be a minor hassle, and make the code a bit harder to follow.  And I realize that `read_surfrad` uses `= open(...)` too.  Not asking for this to be changed, just wanted to point it out.

But: if we're using `open` outside of a context manager, we need to call `file_buffer.close()` when we're done using it (after the call to `pd.read_fwf`)